pkg/operator/controller/certificate/default_cert.go

```diff
@@ -130,3 +140,15 @@ func (r *reconciler) deleteRouterDefaultCertificate(secret *corev1.Secret) (bool
 	}
 	return true, nil
 }
+
+// lookupUserSpecifiedRouterDefaultCertificate checks to see if the given ingress controller's
+// Spec.DefaultCertificate field corresponds to an existing secret. This function assumes that
+// ci.Spec.DefaultCertificate is not nil.
+func (r *reconciler) lookupUserSpecifiedRouterDefaultCertificate(ci *operatorv1.IngressController, namespace string) error {
+	secret := &corev1.Secret{}
+	name := controller.RouterEffectiveDefaultCertificateSecretName(ci, namespace)
+	if err := r.client.Get(context.TODO(), name, secret); err != nil {
```

Comment:
I think it might be best to check if the error is due to a system issue, like a timeout from the api.  If there was a timeout looking up the secret, we wouldn't want to generate a new one.

Reply:
>  If there was a timeout looking up the secret, we wouldn't want to generate a new one.

I think this is the current behavior. If `lookupUserSpecifiedRouterDefaultCertificate` returns an error for any reason (API timeout, the secret does not exist, etc.), then `ensureDefaultCertificateForIngress` returns that error, and the reconcile loop will retry the process again.